tests/loader_test_vectors.rs

```diff
@@ -134,8 +185,74 @@ fn assert_file<T, F: FnOnce() -> IonResult<T>>(skip_list: &[&str], file_name: &s
 
 #[test_resources("ion-tests/iontestdata/good/**/*.ion")]
 #[test_resources("ion-tests/iontestdata/good/**/*.10n")]
-fn good(file_name: &str) {
-    assert_file(ALL_SKIP_LIST, file_name, || load_file(&loader(), file_name));
+fn good_roundtrip_text_binary(file_name: &str) {
+    assert_file(ALL_SKIP_LIST, file_name, || {
+        assert_three_way_round_trip(
+            file_name,
+            |slice| Format::Text(TextKind::Compact).try_dumper_for_slice(slice),
+            |slice| Format::Binary.try_dumper_for_slice(slice),
+        )
+    });
+}
+
+#[test_resources("ion-tests/iontestdata/good/**/*.ion")]
+#[test_resources("ion-tests/iontestdata/good/**/*.10n")]
+fn good_roundtrip_binary_text(file_name: &str) {
+    assert_file(ALL_SKIP_LIST, file_name, || {
+        assert_three_way_round_trip(
+            file_name,
+            |slice| Format::Binary.try_dumper_for_slice(slice),
+            |slice| Format::Text(TextKind::Compact).try_dumper_for_slice(slice),
+        )
+    });
+}
+
+#[test_resources("ion-tests/iontestdata/good/**/*.ion")]
+#[test_resources("ion-tests/iontestdata/good/**/*.10n")]
+fn good_roundtrip_text_pretty(file_name: &str) {
+    assert_file(ALL_SKIP_LIST, file_name, || {
+        assert_three_way_round_trip(
+            file_name,
+            |slice| Format::Text(TextKind::Compact).try_dumper_for_slice(slice),
+            |slice| Format::Text(TextKind::Pretty).try_dumper_for_slice(slice),
+        )
+    });
+}
+
+#[test_resources("ion-tests/iontestdata/good/**/*.ion")]
+#[test_resources("ion-tests/iontestdata/good/**/*.10n")]
+fn good_roundtrip_pretty_text(file_name: &str) {
+    assert_file(ALL_SKIP_LIST, file_name, || {
+        assert_three_way_round_trip(
+            file_name,
+            |slice| Format::Text(TextKind::Pretty).try_dumper_for_slice(slice),
+            |slice| Format::Text(TextKind::Compact).try_dumper_for_slice(slice),
```

Comment:
I don't think I've seen many instances of `try_NOUN` in constructor names. (Though I think most constructors are infallible, which may be why.)

When this is renamed per #243, how would you feel about something like: 
```rust
fn element_writer_for(O: output) -> IonResult<ElementWriter<Output=O>> where O: io::Write
```
?

(As mentioned in another comment, I'm unsure why constructing an `ElementWriter` impl without doing any serialization operations could fail.)

Reply:
See above for my comment around the fallibility.

As for the generic Output, I am not sure how to make this work for the simple slice case when the underlying API is not generic over `Write`.  I am not opposed to this API, I just don't quite know how to make it work for this constrained case (the caller chooses `O` so the implementation would have to be generic for all possible `O: io::Write`, and I cannot do this with the current implementation in Ion C without implementing #49).

Reply:
Added #251 for the naming question.